### PR TITLE
feat: enable cors

### LINKED ISSUE #432: Enable CORS

### PR DESCRIPTION
**Pull request by Oleksii-Klimov**

### CODE REVIEW

**Comment by Oleksii-Klimov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-433.nightly-test.deltixhub.io
>E2E tests status: success

File: src/main/java/com/epam/aidial/core/Proxy.java [hunk 1]
```diff
@@ -121,12 +122,19 @@ private void handleError(Throwable error, HttpServerRequest request) {
      * Called when proxy received the request headers from a client.
      */
     private void handleRequest(HttpServerRequest request) {
+        enableCors(request);
```
Comment (astsiapanay) [inline src/main/java/com/epam/aidial/core/Proxy.java:125]: I think that's not a full implementation of CORS handler. Please take a look at a spec https://developer.mozilla.org/en-US/docs/Glossary/Preflight_request#

You can review the implementation in vertex.io io.netty.handler.codec.http.cors.CorsHandler

**Review by astsiapanay — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/aidial/core/Proxy.java (hunk 1 above)
Comment (Oleksii-Klimov) [reply src/main/java/com/epam/aidial/core/Proxy.java:125]: The HTTP code 204 doesn't seem to be necessary. Currently, we do not use any successful code other than 200 in the application. Additionally, the Spring implementation of CORS also returns code 200 for OPTIONS requests. I'd keep it similar for now.

**Comment by Oleksii-Klimov** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-433.nightly-test.deltixhub.io
>E2E tests status: success

**Review by astsiapanay — APPROVED** [review]

**Pull request merged**